plugins/uefi-dbx/meson.build

```diff
@@ -1,17 +1,33 @@
 cargs = ['-DG_LOG_DOMAIN="FuPluginUefiDbx"']
 plugins += {meson.current_source_dir().split('/')[-1]: true}
 
+snapd_opt = get_option('plugin_uefi_dbx_snapd')
+
+if snapd_opt.require(libcurl.found(),
+  error_message: 'curl is needed for plugin_uefi_dbx_snapd').disable_auto_if(host_machine.system() != 'linux').allowed()
+endif
```

Comment:
I think you've got the logic wrong here still because the allowed isn't stored to `snapd_opt` when compiling for non-Linux.

I _think_ this is what you're looking for:

```c
snapd_opt =get_option('plugin_uefi_dbx_snapd').disable_auto_if(host_machine.system() != 'linux').allowed()
if snapd_opt.require(libcurl.found(),
  error_message: 'curl is needed for plugin_uefi_dbx_snapd').allowed()
endif
```

Reply:
hm `allowed()` returns `bool`, is this what you meant?
```meson
snapd_opt =get_option('plugin_uefi_dbx_snapd').disable_auto_if(host_machine.system() != 'linux')
if snapd_opt.require(libcurl.found(),
  error_message: 'curl is needed for plugin_uefi_dbx_snapd').allowed()
endif
...
if snapd_opt.allowed()
...
```

Reply:
Yeah.

Hmm how about just optimize then like this to only have the one allowed check? 

```
snapd_opt =get_option('plugin_uefi_dbx_snapd').disable_auto_if(host_machine.system() != 'linux')
if snapd_opt.require(libcurl.found(),
  error_message: 'curl is needed for plugin_uefi_dbx_snapd').allowed()
...
endif # snapd_opt.require().allowed()
```

Reply:
There's a separate question whether this should be `.allowed()` or to support auto or rather `.enabled()`

Reply:
> There's a separate question whether this should be .allowed() or to support auto or rather .enabled()

Well we could also sidestep this whole thing by dropping the meson option.  I'm of the opinion that we should just detect snapd automatically in the uefi-dbx plugin, no need to make it a compile option.

@hughsie thoughts on this though?

Reply:
I think I'd rather it be simpler -- so maybe we drop the option?

Reply:
> I think I'd rather it be simpler -- so maybe we drop the option?

and make curl a hard requirement?